### DISCUSSION TITLE
Why does the focus state stay on button click and not a link?

### DISCUSSION THREAD

**Discussion by da-kicks-87** [twbs/bootstrap#32997] [Q&A]
Hello,

Why does the focus state stay on a button  click and not a link click? This seems to be BS specific since testing in an environment without it does not behave this way.  I don't want the focus state to appear on click for both elements.

Using the following CSS does remove it. However its not good practice to do so since doing it also remove the focus state when trying to select the button with the keyboard.

```
.btn:focus {
  outline: none;
  box-shadow: none;
}
```

Anyone have solutions or technical reason why BS treats buttons this way?

**Comment by XhmikosR** [comment]
@patrickhlauke maybe you can reply this?

**Comment by patrickhlauke** [comment]
we need `:focus-visible` for this. otherwise no, we can't just suppress `:focus` this way, as otherwise there won't be any focus indication for keyboard users.
